mpet/utils.py

```diff
@@ -87,6 +87,66 @@ def get_asc_vec(var, Nvol, dt=False):
     return out
 
 
+def central_diff_bulk(array, Nvol, dx):
+    """Gets central diff for derivatives for use in thermal derivatives (which are split between
+    the individual electrodes) for bulk"""
+    varout = {}
+    for sectn in ["a", "c", "s"]:
+        # If we have information within this battery section
+        if sectn in ["a", "c"]:
+            if sectn in array.keys():
```

Comment:
I attempted to do this previously but I think it's pretty difficult because we need to consider cases where there is no separator, where the separator information should be set to 0 for every point on the separator, or where the electrode for the anode doesn't exist, which actually means that we should have no data at the anode.